src/App.tsx

```diff
@@ -1,15 +1,96 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface AppProps {
+  debounceDelay?: number;
+  onSelected?: (person: Person) => void;
+}
+
+export const App: React.FC<AppProps> = ({
+  debounceDelay = 300,
+  onSelected,
+}) => {
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [filteredSuggestions, setFilteredSuggestions] = useState<
+    typeof peopleFromServer
+  >([]);
+  const [inputFocused, setInputFocused] = useState<boolean>(false);
+  const [showNoSuggestionsMessage, setShowNoSuggestionsMessage] =
+    useState<boolean>(false);
+  const previousSearchTerm = useRef<string>('');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
```

Comment:
it works but it'd look a bit simpler with `debounce` function usage (and 2 states for typed value - one is immediate and the second is delayed and used to filter people)

At least, read about debounced and how to implement it

FYI